src/cpp/rtps/builtin/discovery/endpoint/EDPUtils.hpp

```diff
@@ -64,9 +64,12 @@ class EDPUtils
             const HistoryAttributes& history_attr,
             bool is_reader)
     {
-        PoolConfig pool_cfg = PoolConfig::from_history_attributes(history_attr);
-        pool->release_history(pool_cfg, is_reader);
-        TopicPayloadPoolRegistry::release(pool);
+        if (nullptr != pool)
```

Comment:
Shared pointers have a bool operator for a reason, please use it.

```suggestion
        if (pool)
```